### PR TITLE
[move][move-ide][move-compiler] Swap `FilesSourceText` to `MappedFiles` in most places

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

This swaps out `FilesSourceText` for `MappedFiles` whenever possible, along with the following changes: 

- The internal implementation of `MappedFiles` is a little more complicated to support some `move-analyzer` behaviors.
- `MappedFiles` now lives in `move-compiler/shared/files.rs` to reflect it being standalone from the diagnostics.
- This also updates the `move-analyzer` to start using the `MappedFiles` instead of maintaining a separate file state.
- The `Position` form had its fields reverted to non-`pub` with getters to better-clarify the value being returned.

## Test plan 

All tests still run.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 1]
```diff
@@ -1155,24 +1155,15 @@ pub fn get_symbols(
...
-        if *is_dep {
-            hasher.update(fhash.0);
-        }
-        let id = files.add(*fname, source.clone());
-        file_id_mapping.insert(*fhash, id);
-        file_name_mapping.insert(*fhash, PathBuf::from(fname.as_str()));
-        let lines: Vec<String> = source.lines().map(String::from).collect();
-        file_id_to_lines.insert(id, lines);
-    }
+    source_files
```
Comment (awelc) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:1163]: Now that with this PR we will be using mapped files from the compiler it's a shame that we read and hash their content here as well. Would it make sense to (and be less expensive?) to do a different validation here (e.g., read and hash last-modified date for each file and compare these rather then the content)?

**Review by awelc — APPROVED** [review]
LGTM other than a comment/suggestion

File: external-crates/move/crates/move-analyzer/src/symbols.rs (hunk 1 above)
Comment (cgswords) [reply external-crates/move/crates/move-analyzer/src/symbols.rs:1199]: Let's do that in a different PR. I have some ideas for making this work.

**Review by tzakian — APPROVED** [review]
Thank you for this! Changes LGTM as well.

**Pull request merged**